### ISSUE TITLE
Move DarkMode toggle into ZooHeader

### ISSUE THREAD

**Issue by seanmiller26** [zooniverse/front-end-monorepo#5571]
## Package
_Choose from the list:_
- [ ] app-content-pages
- [ ] app-project
- [ ] lib-classifier
- [ ] lib-panoptes-js
- [ ] lib-react-components
- [ ] unknown

## Is your feature request related to a problem? Please describe.
The current placement of our Dark Mode toggle is inconsistent. On desktop pages it is placed on the right margin vertically. 

<img width="400" alt="Screen Shot 2023-10-31 at 4 29 13 PM" src="https://github.com/zooniverse/front-end-monorepo/assets/89043489/790eac7f-a55b-409a-aa40-992a6d39676c">
<img width="400" alt="Screen Shot 2023-10-31 at 4 28 56 PM" src="https://github.com/zooniverse/front-end-monorepo/assets/89043489/5940921b-ceaf-4e22-bfc8-32133e0ad24f">

On mobile sizing it gets placed either below the task area (Classify page) or below the content (Team page).

<img width="400" alt="Screen Shot 2023-10-31 at 4 29 39 PM" src="https://github.com/zooniverse/front-end-monorepo/assets/89043489/3168e606-e839-4045-a7af-12f9b5ba3b4e">
<img width="400" alt="Screen Shot 2023-10-31 at 4 28 40 PM" src="https://github.com/zooniverse/front-end-monorepo/assets/89043489/93fde8ba-92ca-47c4-bd59-60edc4a41aed">

And on some pages it is not even present (Zooniverse About pages).

## Describe the solution you'd like
Place the DarkMode toggle in the ZooHeader. This will allow users to get familiar with the placement regardless of which page they're on, plus will aid in future page design decisions.

Here is a quick read on a few options that are out there: [https://dev.to/zetareticoli/dark-light-mode-toggle-a-usability-issue-1gg2](https://dev.to/zetareticoli/dark-light-mode-toggle-a-usability-issue-1gg2) I would love to hear opinions on which route to explore further, especially dealing with accessibility and technical issues. For now here is a first pass:

<img width="800" alt="Screen Shot 2023-10-31 at 4 53 36 PM" src="https://github.com/zooniverse/front-end-monorepo/assets/89043489/a465c8cb-0eac-4430-a4f4-d61d867f0aa0">

<img width="500" alt="Screen Shot 2023-10-31 at 4 56 57 PM" src="https://github.com/zooniverse/front-end-monorepo/assets/89043489/5b3f2a7a-7654-4db8-bd33-1e1a6c632c0e">

**Comment by eatyourgreens** [comment]
The toggle in the header of https://help.zooniverse.org is quite simple. It's part of the Material theme.

**Comment by seanmiller26** [comment]
This looks great! I'd like to make sure these specifics are addressed:

- The icon should be to the immediate left of the user name, with proper padding
- Match the sizing of Notifications and Messages icons (12px)

**Comment by goplayoutside3** [comment]
I tried out 12px and the icon looked super small. Here are some screenshots with the icon dimensions as 14px, so please let me know if that size is okay. Also, the links in the header are different depending on whether you're signed-in, signed-out, or  whether you're on desktop or mobile. The username is not always there.

Where do you want the theme toggle placed in each of the following scenarios? (Note that the general header padding is fixed in https://github.com/zooniverse/front-end-monorepo/pull/5658. A separate PR)


1. Desktop signed-in
<img width="1098" alt="Screenshot 2023-11-21 at 12 18 50 PM" src="https://github.com/zooniverse/front-end-monorepo/assets/23665803/c9057dd5-a7fa-4be5-8d21-364ca8fb5dd4">
- 

2. Desktop signed-out
<img width="1096" alt="Screenshot 2023-11-21 at 12 20 08 PM" src="https://github.com/zooniverse/front-end-monorepo/assets/23665803/676db3a4-a86c-4bb0-ab05-bb22b3c1fafd">
- 

3. Mobile signed-in
<img width="431" alt="Screenshot 2023-11-21 at 12 19 06 PM" src="https://github.com/zooniverse/front-end-monorepo/assets/23665803/697bf09b-6de9-4213-b3b9-a816e74732c8">
- 

4. Mobile signed-out
<img width="429" alt="Screenshot 2023-11-21 at 12 19 27 PM" src="https://github.com/zooniverse/front-end-monorepo/assets/23665803/f611accb-598f-4b5a-8700-dbb2a0cfdcdc">

**Comment by goplayoutside3** [comment]
Thanks - will do!

**Comment by goplayoutside3** [comment]
Closed by https://github.com/zooniverse/front-end-monorepo/pull/5669
